### PR TITLE
Created api call to find chains near a location

### PR DESCRIPTION
**Pull request by NeedADispenserHere**
API endpoint is /v2/chain/near?longitude=x&latitude=y&radius=z

### CODE REVIEW

File: server/internal/controllers/chain.go [hunk 1]
```diff
@@ -229,6 +229,52 @@ func ChainGetAll(c *gin.Context) {
 	c.JSON(200, chainsJson)
 }
 
+func ChainGetNear(c *gin.Context) {
+	db := getDB(c)
+
+	var query struct {
+		Latitude  float32 `form:"latitude" binding:"required,latitude"`
+		Longitude float32 `form:"longitude" binding:"required,longitude"`
+		Radius    float32 `form:"radius" binding:"required,longitude"`
```
Comment (lil5) [inline server/internal/controllers/chain.go:238]: Remove the longitude from the binding

File: server/internal/controllers/chain.go [hunk 2]
```diff
@@ -229,6 +229,52 @@ func ChainGetAll(c *gin.Context) {
...
+	if err := c.ShouldBindQuery(&query); err != nil && err != io.EOF {
+		c.String(http.StatusBadRequest, err.Error())
+		return
+	}
+
+	chains := []models.Chain{}
+	sql := "SELECT * FROM chains"
+	args := []any{}
+
+	sql = fmt.Sprintf("%s WHERE %s <= ? ", sql, sqlCalcDistance("chains.latitude", "chains.longitude", "?", "?"))
```
Comment (lil5) [inline server/internal/controllers/chain.go:249]: Filter out unpublished loops

File: server/internal/controllers/chain.go [hunk 3]
```diff
@@ -229,6 +229,52 @@ func ChainGetAll(c *gin.Context) {
...
+			"name":                chain.Name,
+			"description":         chain.Description,
+			"address":             chain.Address,
+			"latitude":            chain.Latitude,
+			"longitude":           chain.Longitude,
+			"radius":              chain.Radius,
+			"sizes":               chain.Sizes,
+			"genders":             chain.Genders,
+			"published":           chain.Published,
+			"open_to_new_members": chain.OpenToNewMembers,
```
Comment (lil5) [inline server/internal/controllers/chain.go:271]: To show a list of loops that are nearby only the name, genders and uid (for links) is required you can also sharpen the `select *`

**Review by lil5 — CHANGES_REQUESTED** [review]
Thanks for the PR, some change requests - I'm in the train 🚊 atm I'll test the `where` location tomorrow.

**Comment by NeedADispenserHere** [comment]
I've applied the requested changes

**Comment by lil5** [comment]
Accepted I tried it out in Bruno and it worked great!

**Pull request merged**